app/services/provider_data_api_client.rb

```diff
@@ -0,0 +1,47 @@
+class ProviderDataApiClient
+  class << self
+    def contract_active?(office_code)
+      query(
+        "provider-office/#{office_code}/office-contract-details",
+        200 => ->(_) { true },
+        204 => ->(_) { false },
+      )
+    end
+
+    def user_details(email)
+      query(
+        "provider-users/exists?userEmail=#{email}",
+        200 => ->(data) { data['user'] }
+      )
+    end
+
+    def user_office_details(user_login)
+      query(
+        "provider-users/#{user_login}/provider-offices",
+        200 => ->(data) { data['officeCodes'] },
+        204 => ->(_) { [] },
+      )
+    end
+
+    private
+
+    def query(endpoint, return_values)
```

Comment:
Is there a reason `return_values` can't be a map? if the value is a function call and return it, otherwise return it

Reply:
Good shout, it's barely more code inside `query`, and makes calling it much cleaner.